### ISSUE TITLE
[Bug]: Unable to create any private entity in JS module

### ISSUE THREAD

**Issue by subrata71** [appsmithorg/appsmith#32757]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

When I try to create any private JS object or query or API inside a JS module, it throws an error as shown in the screenshot

<img width="1445" alt="Screenshot 2024-04-18 at 11 25 38 AM" src="https://github.com/appsmithorg/appsmith/assets/3524599/002e3c8d-9877-4dab-8013-f8c0177955bf">

### Steps To Reproduce

1. Create a JS module
2. Add a query
3. It will throw an error

### Environment

Production

### Severity

Blocker to Building

### Version

Self Hosted

**Comment by subrata71** [comment]
## RCA
TL;DR
The issue stemmed from this [line](https://github.com/appsmithorg/appsmith/pull/32448/files#diff-6d72674578cc06e8fc170efb6ab9bf63aff7fe4b2c645edc74b47ad5d9bdeda1R57).

The purpose of [this PR](https://github.com/appsmithorg/appsmith/pull/32448) was to enhance control over permissible fields in both request-body and response-body. Fields annotated with `@JsonView(FromRequest.class)` are deemed permissible in the request body, meaning the deserialization process upon server request considers these fields. Conversely, any field lacking this annotation will be disregarded during deserialization.

In this specific case, the `moduleId` field was transmitted from the client-side in the request body. However, on the server-side, this field lacked the `@JsonView(FromRequest.class)` annotation. Consequently, during deserialization, the absence of `moduleId` led to validation failure.
